utils/navigation.js

```diff
@@ -0,0 +1,79 @@
+import NavigationItem from './NavigationItem';
+import Page from './Page';
+
+function buildMenu({ navigationItem, pathToActiveNav, page, depth }) {
+  const hasChildren = page.hasChildrenAtNavDepth(depth);
+  const titleAtDepth = page.titleAtNavDepth(depth);
+  let item = navigationItem.findItemByTitle(titleAtDepth);
```

Comment:
@Ray2126 There's a bit of indirection here where we're doing a search outside the class-then-insert. I'd find it more intuitive to have a `Navigation.insert` method that encapsulates searching for the right position AND inserts the `NavigationItem` - what do you think?

Reply:
While it's inserting `NavigationItems` the `insert` method could just insert any intermediate nodes that don't exist yet on the way down

Reply:
I don't understand what you mean. Can we catch up about this?

Reply:
Caught up with Tom, we will add an `insert` method on the `Navigation` class.